core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -490,11 +490,6 @@ public MetaData getSnapshotMetaData(SnapshotInfo snapshot, List<IndexId> indices
 
     @Override
     public SnapshotInfo getSnapshotInfo(final SnapshotId snapshotId) {
-        if (getRepositoryData().getIncompatibleSnapshotIds().contains(snapshotId)) {
-            // an incompatible snapshot - cannot read its snapshot metadata file, just return
-            // a SnapshotInfo indicating its incompatible
-            return SnapshotInfo.incompatible(snapshotId);
-        }
```

Comment:
Since `getSnapshotInfoInternal` (just below) is only used by `getSnapshotInfo`, we can move the code in `getSnapshotInfoInternal` directly into `getSnapshotInfo` and get rid of `getSnaphotInfoInternal`